game.cc

```diff
@@ -303,20 +354,250 @@ bool Game::isCheck(int player){
 		return false;
 
 	}
-	else return false;
 }
 
-bool Game::makeMove(int startX, int startY, int endX, int endY){
+bool Game::makeMove(int startX, int startY, int endX, int endY, bool promote = true, bool checkForCheck = true){
```

Comment:
Instead of having a promote flag, let's have a promote character which I can fill from the controller.